internal/types/types.go

```diff
@@ -0,0 +1,66 @@
+package types
+
+import (
+	"errors"
+
+	"github.com/argoproj-labs/argo-cloudops/internal/validations"
+)
+
+type Target struct {
```

Comment:
What if there was a `target` package instead of a `types` package? Just thinking in case we have to add another "type". The `target` package would then have a `Properties` types. I don't know if having a `target.Target` is idiomatic, but if not,  something like `target.Record` is an idea.

@bkono I think we talked about this a while back. You have any thoughts?

Reply:
I was thinking we would also want a project `type` in the future as well. Originally talked with @thbishop-intuit and he was thinking types, but i'm good with changing it to something else. I don't really like `target.Target` since it stutters, but open to ideas.